### PR TITLE
Added New Settings for skip_busy_buffer_layers

### PR DESCRIPTION
**Pull request by kmr-rohit**
Resolves Issue #70584 

### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Adds setting to skip busy buffer layers which are locked during read.

### CODE REVIEW

File: src/Core/Settings.cpp [hunk 1]
```diff
@@ -5808,7 +5808,8 @@ Experimental data deduplication for SELECT queries based on part UUIDs
 )", 0) \
     M(Bool, implicit_select, false, R"(
 Allow writing simple SELECT queries without the leading SELECT keyword, which makes it simple for calculator-style usage, e.g. `1 + 2` becomes a valid query.
-)", 0)
+)", 0) \
+    M(SettingBool, skip_busy_buffer_layers, false, "Skip busy buffer layers during read operations")
```
Comment (alexey-milovidov) [inline src/Core/Settings.cpp:5812]: The docs should be more detailed.

Comment (kmr-rohit) [reply src/Core/Settings.cpp]: Updated Description

**Comment by alexey-milovidov** [comment]
@vmihailenco, yes, it does not need to hold the exclusive lock during reading - it can be made in a similar way to the Memory table engine.

**Comment by kmr-rohit** [comment]
So, Do we are in need of this setting to skip busy layers ? 
@alexey-milovidov

**Comment by alexey-milovidov** [comment]
Thank you for the pull request!
The issue was never a problem for me, though.

**Comment by vmihailenco** [comment]
@alexey-milovidov just want to point out that:

1. First you say "not a problem at all"
2. Then you say "You can implement a setting to skip busy buffer layers, this will be very **appreciated**"
3. Then you ignore the PR
4. And then it is not a problem for **you**.

I have no idea in the end.

**Pull request closed without merging**